File: 백준/Gold/4969. 월요일－토요일/월요일－토요일.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int[] prime = new int[300001];
        prime[1] = 1;

        for (int i = 2; i < 300001; i++) {
            if (prime[i] == 1) {
                continue;
            }

            if (i % 7 == 1 || i % 7 == 6) {
                for (int j = i+i; j < 300001; j +=i) {
                    prime[j] = 1;
                }
            } else {
                prime[i] = 1;
            }
        }

        while (true) {
            int N = Integer.parseInt(br.readLine());

            if (N == 1) {
                break;
            }

            ArrayList<Integer> pf = new ArrayList<>();

            for (int i = 1; i < Math.sqrt(N)+1; i++) {
                if (N % i == 0) {
                    if (prime[i] == 0) {
                        pf.add(i);
                    }

                    if (prime[N/i] == 0 && i != N/i) {
                        pf.add(N/i);
                    }
                }
            }

            pf.sort(Comparator.naturalOrder());

            System.out.print(N + ": ");
            for (int i : pf) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }
}

File: 백준/Gold/4969. 월요일－토요일/월요일－토요일.py
# 월요일-토요일
import sys
input = sys.stdin.readline

prime = [0] * (300001)
prime[1] = 1

for i in range(2, 300001):
    if prime[i] == 1:
        continue
    
    if i % 7 == 1 or i % 7 == 6:
        for j in range(i+i, 300001, i):
            prime[j] = 1
    else:
        prime[i] = 1

while True:
    N = int(input())

    if N == 1:
        break

    pf = set()
    
    for i in range(1, int(N**0.5)+1):
        if N % i == 0:
            if prime[i] == 0:
                pf.add(i)

            if prime[N//i] == 0:
                pf.add(N//i)

    pf = list(pf)
    pf.sort()
    
    print(str(N) + ":", *pf )

    
